include/signer/noscrypt_signer.hpp

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+#include <plog/Init.h>
+#include <plog/Log.h>
+#include <noscrypt.h>
+
+#include "service/nostr_service_base.hpp"
+#include "signer/signer.hpp"
+
+namespace nostr
+{
+namespace signer
+{
+class NoscryptSigner : public INostrConnectSigner
+{
+public:
+    NoscryptSigner(
+        std::shared_ptr<plog::IAppender> appender,
+        std::shared_ptr<nostr::service::INostrServiceBase> nostrService
+    );
+
+    ~NoscryptSigner();
+
+    void receiveConnection(std::string connectionToken) override;
+
+    std::string initiateConnection(
+        std::vector<std::string> relays,
+        std::string name,
+        std::string url,
+        std::string description
+    ) override;
+
+    std::shared_ptr<std::promise<bool>> sign(std::shared_ptr<data::Event> event) override;
+
+private:
+    static constexpr int _nostrConnectKind = 24133; // Kind 24133 is reserved for NIP-46 events.
+
+    Encryption _nostrConnectEncryption;
+
+    std::shared_ptr<NCContext> _noscryptContext;
+    std::shared_ptr<nostr::service::INostrServiceBase> _nostrService;
+
+    ///< Local nsec for communicating with the remote signer.
+    std::shared_ptr<NCSecretKey> _localPrivateKey;
+
+    ///< Local npub for communicating with the remote signer.
+    std::shared_ptr<NCPublicKey> _localPublicKey;
+
+    ///< The npub on whose behalf the remote signer is acting.
+    std::shared_ptr<NCPublicKey> _remotePublicKey;
+
+    ///< An optional secret value provided by the remote signer.
+    std::string _bunkerSecret;
+
+    ///< A list of relays that will be used to connect to the remote signer.
+    std::vector<std::string> _relays;
+
+    #pragma region Private Accessors
+
+    inline std::string _getLocalPrivateKey() const;
+
+    inline void _setLocalPrivateKey(const std::string value);
+
+    inline std::string _getLocalPublicKey() const;
+
+    inline void _setLocalPublicKey(const std::string value);
+
+    inline std::string _getRemotePublicKey() const;
+
+    inline void _setRemotePublicKey(const std::string value);
+
+    #pragma endregion
+
+    #pragma region Setup
+    
+    /**
+     * @brief Initializes the noscrypt library context into the class's `context` property.
+     */
+    void _initNoscryptContext();
```

Comment:
Dead function. Moved to constructor.

It looks like I forgot to remove these functions from the header.

Reply:
This function is invoked in the version of the constructor used on this branch.  Perhaps your latest work hasn't been merged to `master`?

Reply:
See other comment